File: version/1.8.9/src/main/java/com/github/sorusclient/client/adapter/v1_8_9/ServerImpl.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter.v1_8_9;

import com.github.sorusclient.client.adapter.IServer;
import v1_8_9.net.minecraft.client.network.ServerInfo;

public class ServerImpl implements IServer {

    private final ServerInfo serverInfo;

    public ServerImpl(ServerInfo serverInfo) {
        this.serverInfo = serverInfo;
    }

    @Override
    public String getIP() {
        return serverInfo.address;
    }

}

File: version/1.8.9/src/main/kotlin/com/github/sorusclient/client/adapter/v1_8_9/ServerImpl.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter.v1_8_9

import com.github.sorusclient.client.adapter.IServer
import v1_8_9.net.minecraft.client.network.ServerInfo

class ServerImpl(private val serverInfo: ServerInfo) : IServer {
    override val ip: String
        get() = serverInfo.address
}